lib/cn/omf.h

```diff
@@ -226,21 +229,35 @@ OMF_SETGET(struct wbt_lfe_omf, lfe_kmd, 16)
 
 /*****************************************************************
  *
- * Vblock header OMF
+ * Vblock footer OMF
  *
  ****************************************************************/
 
-#define VBLOCK_HDR_MAGIC ((u32)0xea73feed)
+#define VBLOCK_FOOTER_MAGIC ((u32)0xea73feed)
+#define VBLOCK_FOOTER_LEN   4096
 
-/* Version 2 header */
-struct vblock_hdr_omf {
-    uint32_t vbh_magic;
-    uint32_t vbh_version;
-    uint64_t vbh_vgroup;
+/*
+ * min_key is stored at offset VBLOCK_FOOTER_LEN - (2 * HSE_KVS_KEY_LEN_MAX)
+ * min key is inclusive in all the vblocks
+ *
+ * max_key is stored at offset VBLOCK_FOOTER_LEN - HSE_KVS_KEY_LEN_MAX
+ * max key is exclusive in all but the last vblock
+ */
+struct vblock_footer_omf {
+    uint32_t vbf_magic;
+    uint32_t vbf_version;
+    uint64_t vbf_vgroup;
+    uint32_t vbf_min_klen;
```

Comment:
Maybe just use a uint16_t for key lengths?

Reply:
In any case static assert that KEY_LEN_MAX <= UINTX_MAX

Reply:
Picked the type from kblock_hdr_omf. Have you changed it to uint16_t in your PR for the kblock header?

Reply:
I have not. @alexttx maybe we should create a ticket about tweaking OMF for little things like this prior to 3.0? Would be nice to start jotting down little ideas here and there.

Reply:
What other tweaks do you have in mind?
Regarding lengths, let's change them to uint16_t in these PRs.

Reply:
Changed the omf klen to uint16_t.

Reply:
Nothing off the top of my head except for this at the moment. Will give it more thought

Reply:
I'll submit a PR to fix the kblock. uint32_t -> uint16_t